### PR TITLE
has shared link option

### LINKED ISSUE #93: Build options for shared library
The CML needs to offer a shared library option, and it should be possible to build the test and tool executables using dynamic linking. And we can test those configurations in the actions.

### PR DESCRIPTION
**Pull request by klemens-morgenstern**
Closes #93.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,6 +34,16 @@ project(
 #set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_RELWITHDEBINFO ON CACHE STRING "")
 
 option(MRDOX_BUILD_TESTS "Build tests" ON)
+option(MRDOX_BUILD_SHARED "Link shared" OFF)
+
+if (MRDOX_BUILD_SHARED)
```
Comment (vinniefalco) [inline CMakeLists.txt:39]: is this how its usually done with cmake? I mean, is there not some kind of standard switch or something?

**Comment by vinniefalco** [comment]
I think maybe I did not write the issue up correctly... MrDox has to have a plugin system (i.e. load DLLs / shared libraries dynamically at runtime) so that people can author their own generators without having to build MrDox (and LLVM with it). Or at the least so they don't need to fork the program. 

For this to work they need to be able to call in to MrDox functions and I am assuming this has to happen via shared linking...

**Comment by klemens-morgenstern** [comment]
#58 is the one.

**Comment by vinniefalco** [comment]
> That's a different issue altogether I think. I'll look into it.

Yes, building the plugin architecture is a issue number 58, but this issue is about setting up the build system for it (and having it tested). Most of this pull request looks useful; we need those macros properly set for non-windows platforms, and we need the switches in the CML..

When mrdox is installed from a package the tool should come as a single executable not as an executable with `main` plus the mrdox-api as a shared DLL. But it should still have exports or whatever, so it can load the plugins at runtime, and so those plugins can import from the executable and call into mrdox.

For the tests, who cares. Maybe we should make the mrdox-test link to mrdox dynamically so that CI can test it?

**Pull request closed without merging**